integration/client.go

```diff
@@ -69,6 +69,23 @@ func (c client) queryRaw(data []byte) (rawQueryResults, error) {
 	return results, nil
 }
 
+func (c client) queryGroupBy(data []byte) (groupByQueryResults, error) {
```

Comment:
Can you pull out the common parts and move it to a separate `doQuery` function and reuse it in `queryRaw` and `queryGrouped`?